integration_test/playertesting/PlayerTesting.ts

```diff
@@ -46,6 +110,22 @@ export const expectEvents = async (
   );
 };
 
+/**
+ * Calls the given function with the player instance and expects the given event to occur.
```

Comment:
> Starts listening for the specified `SingleEventExpectationz before executing the passed `fn` function.
This is the race-condition-safe version of calling `callPlayer` and `expectEvent` after that. Useful when events are directly tied to calls in the `fn` function block and therefore synchronously emitted.

Maybe more in that direction.

Reply:
fixed in a4169682